sql/analyzer/pushdown.go

```diff
@@ -887,3 +887,94 @@ func stripDecorations(ctx *sql.Context, a *Analyzer, node sql.Node, scope *Scope
 		}
 	})
 }
+
+func replacePkSort(ctx *sql.Context, a *Analyzer, n sql.Node, scope *Scope, sel RuleSelector) (sql.Node, transform.TreeIdentity, error) {
+	return transform.NodeWithCtx(n, nil, func(tc transform.Context) (sql.Node, transform.TreeIdentity, error) {
+		n := tc.Node
+
+		// Find order by nodes
+		s, ok := n.(*plan.Sort)
+		if !ok {
+			return n, transform.SameTree, nil
+		}
+
+		// Must be sorting by ascending
+		for _, field := range s.SortFields {
+			if field.Order != sql.Ascending {
+				return n, transform.SameTree, nil
+			}
+		}
+
+		// Only replace iff immediate child is ResolvedTable
+		rs, ok := s.UnaryNode.Child.(*plan.ResolvedTable)
+		if !ok {
+			return n, transform.SameTree, nil
+		}
+
+		// Extract primary key columns
+		var pkColNames []string
+		for _, col := range rs.Schema() {
+			if col.PrimaryKey {
+				pkColNames = append(pkColNames, col.Name)
+			}
+		}
```

Comment:
Let's get these from the `Expressions()` of the "PRIMARY" index, since those will be in order.